module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -0,0 +1,579 @@
+            throw new ILSException("500: Internal Server Error");
+        default:
+            return json_decode($response->getBody());
+        }
+    }
+
+    /**
+     * Get raw object of item from inventory/items/
+     *
+     * @param string $itemId Item-level id
+     *
+     * @return array
+     */
+    public function getStatus($itemId)
+    {
+        $item = $this->getItem($itemId);
+        return $item->status;
+    }
+
+    /**
+     * This method calls getStatus for an array of records or implement a bulk method
+     *
+     * @param array $idList Item-level ids
+     *
+     * @return array values from getStatus
+     */
+    public function getStatuses($idList)
+    {
+        $status = [];
+        foreach ($idList as $id) {
+            $status[] = $this->getStatus($id);
+        }
+        return $status;
+    }
+
+    /**
+     * Retrieves renew, hold and cancel settings from the driver ini file.
+     *
+     * @param string $function The name of the feature to be checked
+     * @param array  $params   Optional feature-specific parameters (array)
+     *
+     * @return array An array with key-value pairs.
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getConfig($function, $params = null)
+    {
+        return isset($this->config[$function]) ? $this->config[$function] : false;
+    }
+
+    /**
+     * This method queries the ILS for holding information.
+     *
+     * @param string $itemId      Item-level id
+     * @param array  $patronLogin Patron login information
+     *
+     * @return array An array of associative holding arrays
+     */
+    public function getHolding($itemId, array $patronLogin = null)
+    {
+        $record = $this->getItem($recordId);
+        $holdingResponse = $this->makeRequest(
+            'GET',
+            '/holdings-storage/holdings/' . $record->holdingsRecordId
+        );
+        $holding = json_decode($holdingResponse->getBody());
+        /*
+        $instance = $this->makeRequest(
+            'GET',
+            '/inventory/instances/' . $holding->instanceId
+        );
+        */
+        $location = $record->permenantLocation->name
+            ?? ($holding->electronicLocation->uri ? "Electronic" : null);
+        $locationhref = $holding->electronicLocation->uri ?? null;
+        return [[
+            'id' => $recordId,
+            'availability' => $record->status->name == 'Available',
+            'status' => $record->status->name,
+            'location' => $location,
+            'locationhref' => $locationhref,
+            'barcode' => $record->barcode ?? '',
+            'callnumber' => $holding->callNumber,
+            'notes' => $record->notes ?? '',
+        ]];
+    }
+
+    /**
+     * Patron Login
+     *
+     * This is responsible for authenticating a patron against the catalog.
+     *
+     * @param string $username The patron username
+     * @param string $password The patron password
```

Comment:
Right now it appears that the password is simply being ignored. It might make sense to add a configuration to determine which field is used as a password (i.e. in case somebody wants to validate a user with their last name, like we do in the Voyager driver).

Reply:
Login has now been implemented and is using password